src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -12820,6 +12820,42 @@ static void Main()
                 expectedOutput: "(<>z__ReadOnlyArray<System.Int32>) [1, 2, 0], (<>z__ReadOnlyArray<System.Object>) [1, 2, null], ");
         }
 
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/72539")]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/74676")]
+        public void SynthesizedCollections_EnsureCompilerGenerated()
+        {
+            string source = """
+                using System.Collections.Generic;
+
+                class C
+                {
+                    void M()
+                    {
+                        IEnumerable<int> x = [1];
+                        IEnumerable<int> y = [2, 3];
+                        IEnumerable<int> z = [.. x];
+                    }
+                }
+                """;
```

Comment:
>""" [](http://example.com/codeflow?start=16&length=3)

Let's also verify the complete set of attributes emitted to the assembly. (Sorry I overlooked this earlier.)
```csharp
    Report(x);
    Report(y);
    Report(z);
}
static void Report<T>(IEnumerable<T> e)
{
    var type = e.GetType();
    Console.Write("{0}: ", type.Name);
    foreach (var a in type.GetCustomAttributes())
        Console.Write("{0}, ", a);
    Console.WriteLine();
}
```